### PR TITLE
Test PUSHing an object into the cache and then GETting it.

### PR DESCRIPTION
**Pull request by ywkaras**
With a few variations on the client connection protocol.

The new test can also be modified temporarily and used to test client-side communication when GETting multi-GB objects.

### CODE REVIEW

**Comment by bryancall** [comment]
@ywkaras How large is the object and it looks like the test is enabled by default?  Just verifying because I thought in the standup meeting we had it was going to be disabled.  Depending on how long it takes would determine if it should be enabled by default.

**Comment by ywkaras** [comment]
In the checked-in version of the test, the object is 10MB, and is downloaded 4 times:
1.  IPv4 cleartext HTTP 1.1
2. IPv4 TLS HTTP 1.1
3. IPv4 TLS HTTP 2
4. IPv6 TLS HTTP 2

Also I think this is the only test that exercises the PUSH method.  It only takes a few seconds to run in docker on my MacBook.

File: tests/gold_tests/bigobj/bigobj.test.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+#  limitations under the License.
+
+Test.Summary = '''
+Test client-side communication when GETting a very large object.
+'''
+
+Test.SkipUnless(
+    Condition.HasCurlFeature('http2'),
+    Condition.HasProgram('nc', 'nc (netcat) command needed for this test'),
+    Condition.HasProgram('cc', 'cc (C compiler) command needed for this test')
```
Comment (randall) [inline tests/gold_tests/bigobj/bigobj.test.py:27]: Back in `6c1a90f4f209a02cf1345ac4c20e2e67c4c58db1`, it was decided not to check for things like nc.

And in the other tests that build custom c code, we don't test for cc.

Comment (ywkaras) [reply tests/gold_tests/bigobj/bigobj.test.py]: OK changed.

**Pull request merged**